tools/src/bin/web-server.rs

```diff
@@ -94,20 +96,47 @@ fn handle(
     }
 
     let tree_name = &path[0];
-    let kind = &path[1];
+    let kind = path[1];
 
     println!("DBG {:?} {} {}", path, tree_name, kind);
 
-    match &kind[..] {
-        "rev" => {
+    match kind {
+        "rev" | "hgrev" => {
             if path.len() < 3 {
                 return not_found();
             }
 
-            let rev = &path[2];
+            let mut rev = path[2].to_string();
             let path = path.clone().split_off(3);
             let path = path.join("/");
 
+            if kind == "hgrev" {
+                let tree_config = cfg.trees.get(*tree_name).unwrap();
+                let git_path = match config::get_git_path(tree_config) {
+                    Ok(git_path) => git_path,
+                    Err(_) => return not_found(),
+                };
+                let output_result = Command::new("git")
+                    .arg("cinnabar")
+                    .arg("hg2git")
+                    .arg(&rev)
+                    .current_dir(&git_path)
+                    .output();
+                match output_result {
+                    Ok(output) if output.status.success() => {
+                        rev = git_ops::decode_bytes(output.stdout).trim().to_string();
```

Comment:
I wonder if it'd be better to just redirect to the git revision... Stuff like the permalink things / blame / etc will still point to git revs...

Would also potentially be nice to have in the menu a "HG link" or something, but that can be future work or what not.

Reply:
If our goal is to maintain the git revisions as our canonical revisions, then I strongly agree here that /hgrev/ should likely just be a permanent redirect.  If we want to move mozilla-central and other hg-using repositories over to being primarily addressed using hg revisions, then I think it makes to possibly support both, but strongly bias such repositories to using hg.

Given that the people previously most focused on enhancing mercurial are no longer at Mozilla (ex: :gps), and that the CVS history of mozilla-central cannot be expressed in hg revisions (I think?), and that mercurial periodically frustrates me personally (the most important heuristic! ;), it seems like it probably makes sense to continue being git-centric for now.  (And therefore doing the redirect.)

Reply:
In terms of the HG link, I agree that that seems like a nice affordance.  That seems like it probably wants to be a separate bug and we should probably reach consensus around the UX for the "I probably just want the URL in my clipboard" use-case and conventions there I discuss in https://github.com/mozsearch/mozsearch/pull/368#issuecomment-674979638 for Emilio's patch for that.

Reply:
Yeah, I plan to look at that soonish :)